### PR TITLE
refactor phq stepping

### PR DESCRIPTION
**Pull request by ericcf**

### CODE REVIEW

File: app/models/phq_stepping_assessment.rb [hunk 1]
```diff
@@ -88,15 +88,9 @@ def fill_in_unanswered_questions(assessment)
...
-    return - 1 if @date < first_sunday
-
-    last_saturday = @date.end_of_week(:sunday)
-
-    # Week 1 would be (0/7) + 1 = 1, #Week 2 would be (7/7) + 1 = 2, etc
-    (last_saturday - first_saturday) / 7 + 1
+  # Assessments are always made available on the same day of the
+  # week, starting on week 2 of study enrollment.
+  def find_week_of_assessment(study_start_date)
+    ((@date - study_start_date).days / 1.week).floor + 1
```
Comment (michaelwehrley) [inline app/models/phq_stepping_assessment.rb]: Nice improvement

Comment (michaelwehrley) [inline app/models/phq_stepping_assessment.rb]: couldn't you do `.ceil` and not have to add 1.  For instance: `((@date - study_start_date).days / 1.week).ceil`

File: app/views/think_feel_do_engine/coach/patient_dashboards/index.html.erb [hunk 2]
```diff
@@ -3,21 +3,18 @@
 <div class="btn-toolbar" role="toolbar">
   <%= link_to "Group", (defined?(think_feel_do_dashboard) ? think_feel_do_dashboard.group_path(@group) : "#"), class: "btn btn-default" %>
   <%= link_to @active_patients ? "Inactive Patients" : "Active Patients",
-              patient_button_link(@active_patients),
+              think_feel_do_engine.coach_group_patient_dashboards_path(@group, active: !@active_patients),
```
Comment (michaelwehrley) [inline app/views/think_feel_do_engine/coach/patient_dashboards/index.html.erb]: :bread:

File: app/models/phq_stepping_assessment.rb (hunk 1 above)
Comment (ericcf) [inline app/models/phq_stepping_assessment.rb]: I don't think so - what if `@date == study_start_date`? Then ceil will be 0, but it's week 1.

Comment (michaelwehrley) [inline app/models/phq_stepping_assessment.rb]: :gotcha

**Comment by michaelwehrley** [comment]
Nice improvements.

**Pull request merged**